### PR TITLE
Datasets: Add Polars Integration 

### PR DESCRIPTION
**Pull request by c-peters**
This PR is now ready for review

### CODE REVIEW

**Comment by severo** [comment]
OK. The CI breaks because the brackets are interpreted by Svelte? Not sure what is the correct way of escaping them? `\{` and `\}`maybe?

> A hugging face url can be constructed from the `username` and `dataset` name like this `hf://datasets/{username}/{dataset}/{path_to_file}`.

**Comment by c-peters** [comment]
Moved the section out of `<Tip></Tip>` assuming that was the problem

File: docs/hub/datasets-libraries.md
Comment (albertvillanova) [inline docs/hub/datasets-libraries.md:16]:
```suggestion
| [Polars](./datasets-polars) |  A DataFrame library on top of an OLAP query engine.                                                                                                                    | ✅ | ✅ |
```

File: docs/hub/datasets-polars-operations.md
Comment (lhoestq) [inline docs/hub/datasets-polars-operations.md:5]: maybe you can showcase a more interesting dataset like the recent Common Crawl statistics at https://huggingface.co/datasets/commoncrawl/statistics ?

File: docs/hub/datasets-polars-file-formats.md
Comment (ritchie46) [inline docs/hub/datasets-polars-file-formats.md:5]: Arrow/ IPC as well?

File: docs/hub/datasets-polars-file-formats.md
Comment (ritchie46) [inline docs/hub/datasets-polars-file-formats.md:21]: `scan_csv` should be `read_csv`, given the example.

File: docs/hub/datasets-polars-auth.md
Comment (lhoestq) [inline docs/hub/datasets-polars-auth.md:1]: does Polars support authentication from `huggingface-cli login` ? it stores the token at `~/.cache/huggingface/token` where `~/.cache/huggingface` is the default `HF_HOME`.

File: docs/hub/datasets-polars-file-formats.md
Comment (lhoestq) [inline docs/hub/datasets-polars-file-formats.md:42]: we don't support IPC format yet (only streaming for legacy reasons), maybe let's remove this for now and we can add back once the Hub supports IPC
```suggestion
```

File: docs/hub/datasets-polars-operations.md
Comment (lhoestq) [inline docs/hub/datasets-polars-operations.md:1]: love this page !

File: docs/hub/datasets-polars-optimizations.md
Comment (lhoestq) [inline docs/hub/datasets-polars-optimizations.md:19]:
```suggestion
df = pl.read_csv("hf://datasets/commoncrawl/statistics/tlds.csv", try_parse_dates=True)
```

**Review by lhoestq — COMMENTED** [review]
Cool ! Should there be a section on writing to HF as well ? (it can work using `fsspec` I believe)

I added minor suggestions before we merge:

File: docs/hub/datasets-polars-auth.md
Comment (ritchie46) [reply docs/hub/datasets-polars-auth.md:1]: @nameexhaustion do we support this?

Comment (lhoestq) [reply docs/hub/datasets-polars-auth.md:1]: let's keep things as is for now and we can update later if needed

Comment (nameexhaustion) [reply docs/hub/datasets-polars-auth.md:1]: > @nameexhaustion do/can we support this?

made a PR here https://github.com/pola-rs/polars/pull/17881

**Comment by c-peters** [comment]
Putting this as draft, we need to do a Polars python release on our end first

**Comment by c-peters** [comment]
> Putting this as draft, we need to do a Polars python release on our end first

Release is out

**Comment by c-peters** [comment]
> Cool ! Should there be a section on writing to HF as well ? (it can work using `fsspec` I believe)

Writing is not natively supported at the moment, some parts might work with `fsspec`, but I would like to hold off on the section until we add native support

**Review by lhoestq — APPROVED** [review]
Alright let's merge this then ! :)

**Pull request merged**
